src/js/components/Carousel.js

```diff
@@ -58,6 +70,33 @@ export default class Carousel extends Component {
     scrollParents.forEach(function (scrollParent) {
       scrollParent.removeEventListener('scroll', this._handleScroll);
     }.bind(this));
+
+    this._unmountHammer();
+  }
+
+  _unmountHammer () {
+    if (this.hammer) {
```

Comment:
I didn't need bind the `this` context here because I'm calling `_unmountHammer` and `_updateHammer` directly from inside the same context. Not sure if that answers your question.